src/dmd/delegatize.d

```diff
@@ -86,6 +86,25 @@ private void lambdaSetParent(Expression e, FuncDeclaration fd)
         alias visit = typeof(super).visit;
         FuncDeclaration fd;
 
+        private void setParent(Dsymbol s)
+        {
+            VarDeclaration vd = s.isVarDeclaration();
+            FuncDeclaration pfd = s.parent ? s.parent.isFuncDeclaration() : null;
+            s.parent = fd;
+            if (!vd || !pfd)
+                return;
+            // move to fd's closure when applicable
+            foreach (i; 0 .. pfd.closureVars.dim)
```

Comment:
BTW, one should be careful when using `foreach (; 0 .. max)` and changing the `max` as the emitted code caches it:
https://run.dlang.io/gist/run-dlang/b0b40d58474fb7c35b22803b2432c02f?compiler=dmd

Reply:
Yes, I notice that, but the code here breaks immediately after `remove()` is called.